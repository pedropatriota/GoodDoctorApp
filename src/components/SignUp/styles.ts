import styled from "styled-components"
import { ErrorMessage } from "formik"

export const Container = styled.div`
  width: 40%;
  form {
    display: flex;
    flex-direction: column;
    max-width: 100%;
    gap: 1.5rem;
    label {
      color: #fff;
      font-size: 1.8rem;
    }
    input {
      width: 100%;
      height: 5rem;
      border-radius: 8px;
      border: none;
      margin-top: 0.3rem;
      font-size: 1.6rem;
      padding: 0 1rem;
      color: #444;
    }
    button {
      text-transform: uppercase;
      cursor: pointer;
    }
  }
`
export const Error = styled(ErrorMessage)`
  color: red;
`
