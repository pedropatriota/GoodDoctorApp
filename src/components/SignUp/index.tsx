import * as S from "./styles"
import { Formik, Form, Field, FormikHelpers } from "formik"
import * as Yup from "yup"
import Button from "../shared/Button"

interface Values {
  firstName: string
  lastName: string
  phoneNumber: string
  email: string
}

const initialValues = {
  firstName: "",
  lastName: "",
  phoneNumber: "",
  email: "",
}

const validation = Yup.object({
  firstName: Yup.string()
    .max(15, "Deve conter no máximo 15 caracteres")
    .required("Nome é obrigatório"),
  lastName: Yup.string()
    .max(20, "Deve conter no máxmimo 20 caracteres")
    .required("Sobrenome é obrigatório"),
  phoneNumber: Yup.string()
    .max(11, "Deve conter no máximo 15 caracteres")
    .required("Escreva um número de telefone válido"),
  email: Yup.string().email("E-mail inválido").required("E-mail é obrigatório"),
})

const SignIn = () => (
  <S.Container>
    <Formik
      initialValues={initialValues}
      validationSchema={validation}
      onSubmit={async (
        values: Values,
        { setSubmitting }: FormikHelpers<Values>
      ) => {
        await alert(JSON.stringify(values, null, 2))
        setSubmitting(false)
      }}
    >
      {(formik) => {
        console.log(formik)
        return (
          <Form>
            <Field name="firstName" type="text" placeholder="Nome" />
            <S.Error name="firstName" />

            <Field name="lastName" type="text" placeholder="Sobrenome" />
            <S.Error name="lastName" />

            <Field name="phoneNumber" type="text" placeholder="Telefone" />
            <S.Error name="phoneNumber" />

            <Field name="email" type="email" placeholder="E-mail" />
            <S.Error name="email" />

            <Button width="100%" margin="3rem 0" height="5rem" type="submit">
              Cadastrar
            </Button>
          </Form>
        )
      }}
    </Formik>
  </S.Container>
)

export default SignIn
