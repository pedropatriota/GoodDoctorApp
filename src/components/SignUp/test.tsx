import { render, screen } from "@testing-library/react"
import  SignUp from "."

describe("testing component  SignUp", () => {
  it("Should render the heading", () => {
    const { container } = render(<SignUp />)

    expect(screen.getByRole("heading", { name: /SignUp/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

})
