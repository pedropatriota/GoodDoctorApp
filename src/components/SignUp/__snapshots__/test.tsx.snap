// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component  SignIn Should render the heading 1`] = `
<div
  class="styles__Container-hXcVkP"
>
  <h1>
     SignIn 
  </h1>
</div>
`;
